lgc/builder/BuilderImpl.h

```diff
@@ -583,6 +583,53 @@ class MatrixBuilder : virtual public BuilderImplBase {
   // Create matrix inverse operation.
   llvm::Value *CreateMatrixInverse(llvm::Value *const matrix, const llvm::Twine &instName = "") override final;
 
+  // Cooperative matrix operations.
+  // Enumate the supported binary operation
+  enum class OperatorType : unsigned {
+    IAdd = 128,
+    FAdd,
+    ISub,
+    FSub,
+    IMul,
+    FMul,
+    UDiv,
+    SDiv,
+    FDiv,
+  };
+
+  // Represent the info of a cooperative matrix
+  struct CooperativeMatrixInfo {
+    unsigned elemCount;
+    unsigned subElemCount;
+    llvm::Type *elemType;
+    llvm::Type *subElemType;
+  };
+
+  // Create cooperative matrix load operation.
+  llvm::Value *CreateCooperativeMatrixLoad(llvm::Value *pointer, llvm::Value *stride, llvm::Value *colMajor,
+                                           llvm::Value *alignment, const llvm::Twine &instName = "") override final;
+
+  // Create cooperative matrix store operation.
+  llvm::Value *CreateCooperativeMatrixStore(llvm::Value *pointer, llvm::Value *object, llvm::Value *stride,
+                                            llvm::Value *colMajor, llvm::Value *alignment,
+                                            const llvm::Twine &instName = "") override final;
+
+  // Create cooperative matrix convert operation
+  llvm::Value *CreateCooperativeMatrixConvert(llvm::Value *source, llvm::Value *dest,
+                                              const llvm::Twine &instName = "") override final;
+
+  // Create cooperative matrix binary operation
+  llvm::Value *CreateCooperativeMatrixBinaryOp(llvm::Value *op, llvm::Value *operand1, llvm::Value *operand2,
+                                               const llvm::Twine &instName = "") override final;
```

Comment:
Could `op` just be of type `OperatorType`?

Reply:
`BuilderRecorder::record(BuilderRecorder::Opcode opcode, Type *resultTy, ArrayRef<Value *> args,                                const Twine &instName)` needs the second argument be  an array of  `llvm::Value *`. So the arguments of the generated Create* call should be the same type.

Reply:
My last reply is incorrect. The argument can be of enumator type. I move the `OperatorType` to class `Builder` and rename it `CooperativeMatrixArithOp`.  Change the first argument of `CreateCooperativeMatrixBinaryOp` to type `CooperatvieMatrixArithOp`.